lib/avro/builder/type_factory.rb

```diff
@@ -4,28 +4,33 @@ module Builder
     # This concern is used by classes that create new Type instances.
     module TypeFactory
 
+      SPECIFIC_TYPES = Set.new(%w(Array Enum Fixed Map Record Union).map(&:freeze)).freeze
```

Comment:
The primitive types are handled by a generic types. These "specific" types each have a class defined for them. There's actually a `SpecificType` concern they all include. But I did have trouble naming this, so I'm open to suggestions. Originally I just called this `OTHER_TYPES`.